src/App.test.js

```diff
@@ -1,10 +1,36 @@
 import React from "react";
 import { shallow, mount } from "enzyme";
 import App from "./App";
+
+let app;
+let appInstance;
+let originalState;
+
+beforeEach(() => {
+     app = shallow(<App />);
+     appInstance = app.instance();
+     originalState = JSON.parse(JSON.stringify(app.instance().state));
+});
+
 it("renders without crashing", () => {
   shallow(<App />);
 });
+
 it("snapshot testing that it renders correctly", () => {
   const wrapper = mount(<App />);
   expect(wrapper).toMatchSnapshot();
 });
+
+it("should not change state when authorization response (implicit grant) is not defined", () => {
+  appInstance.handleAuthenticationInfoUpdate();
+  expect(appInstance.state).toEqual(originalState);
```

Comment:
I think this should be toBe. Not only do you want them to be equal, you want it to be literally the same object.

`var obj1 = { key: "value" };`
`var obj2 = { key: "value" };`
`expect(obj1).toBe(obj2); // false`
`expect(obj1).toEqual(obj2); // true`

Reply:
This will not apply because originalState is a copy of appInstance.state. They are not the same objects.

Reply:
Okay.